Automattic/wp-calypso
# Add Publish Confirmation AB test

This PR adds the `postPublishConfirmation` AB test, and combines it with the `post-editor/delta-post-publish-flow` feature flag so that the publish confirmation feature can be turned on/off at will irrespective of the AB test.

This is part of a larger epic (See p3Ex-2ri-p2).

Test details:

• Name: `postPublishConfirmation`
• Date: 2017-07-13
• Variations:
  • `showPublishConfirmation`: (30) A confirmation sidebar will be shown on Publish.
  • `noShowPublishConfirmation`: (Default | 70) No confirmation sidebar is shown on Publish.

To test:

- Check out branch
- Run `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
- Create a new post.
- Once the editor is loaded, a variation for the postPublishConfirmation should be set (`localStorage.setItem('debug', 'calypso:abtests');`).
- Publish the new post.
  - `showPublishConfirmation`: A confirmation sidebar should be shown on Publish.
  - `noShowPublishConfirmation`: No confirmation sidebar is shown on Publish.
- Create a new post.
- Use the dev ABTest tool to set the other variation (or `localStorage.setItem('ABTests','{"postPublishConfirmation_20170713":"showPublishConfirmation"}')` or use `noPublishConfirmation` for the other variation).
- Verify that it works correctly.
